### DISCUSSION TITLE
You Should Rarely Need `eval(parse())` | /en/2023/02/eval-parse/

### DISCUSSION THREAD

**Discussion by giscus** [rbind/yihui#1510] [Comments]
Many programming languages have a function to evaluate a string as code or an expression, e.g., eval() in R. Once you learn it, it becomes tempting to abuse it. Instead of writing code naturally and &hellip;

https://yihui.org/en/2023/02/eval-parse/

**Comment by fncokg** [comment]
I found that in function `ppclust::fcm`, there are many codes like this:
```
compd <- parse(text = paste0(".compdist(x[i,], v[j,], dmetric='", dmetric, "', p=", pw, ")"))
# some other codes
for(j in 1:k)
    for(i in 1:n)
        d[i,j] <- eval(compd)
```
I'm surprised and I wonder why not simply replace those codes with this:
```
for(j in 1:k)
    for(i in 1:n)
        d[i,j] <- .compdist(x[i,],v[j,],dmetric=dmetric,p=pw)
```

**Reply by yihui** [reply]
Indeed there are many `eval()` calls in this package: https://github.com/search?q=repo%3Acran%2Fppclust+eval%28&type=code I think you are right that these `eval(parse())` calls are unnecessary. You may contact its authors and see if we missed anything.
